src/lib.rs

```diff
@@ -125,43 +250,80 @@ impl ElasticCollector {
 }
 
 impl Default for ElasticCollector {
-    fn default() -> ElasticCollector {
+    fn default() -> Self {
         ElasticCollector::new_local(IndexTemplate::default())
     }
 }
 
+fn build_batch(events: &[Event<'static>], template: &IndexTemplate) -> Vec<u8> {
+    let mut buf = Cursor::new(Vec::new());
+    let formatter = RenderedJsonFormatter::new();
+
+    for evt in events {
+        let idx = template.index(&evt.timestamp());
+
+        //Writes a header struct of the form: {"index":{"_index":"{}","_type":"{}"}}\n
+        buf.write_all(b"{\"index\":{\"_index\":\"").unwrap();
+        buf.write_all(idx.as_bytes()).unwrap();
+        buf.write_all(b"\",\"_type\":\"").unwrap();
+        buf.write_all(TYPENAME.as_bytes()).unwrap();
+        buf.write_all(b"\"}}\n").unwrap();
+
+        //Writes the message body to the buffer
+        formatter.write_event(&evt, &mut buf).unwrap();
+        buf.write(b"\n").unwrap();
+    }
+
+    buf.into_inner()
+}
+
+fn build_index_template(template: &IndexTemplate) -> Vec<u8> {
+    let mut buf = Cursor::new(Vec::new());
+
+    //Writes a body like {"template":"testlog-*","mappings":{"emitlog":{"properties":{"@t":{"type":"date","format":"yyyy-MM-ddTHH:mm:SSSZ"}}}}}
+    buf.write_all(b"{\"template\":\"").unwrap();
+    buf.write_all(template.prefix.as_bytes()).unwrap();
+    buf.write_all(b"*\",\"mappings\":{\"").unwrap();
+    buf.write_all(TYPENAME.as_bytes()).unwrap();
+    buf.write_all(b"\":{\"properties\":{\"@t\":{\"type\":\"date\",\"format\":\"yyyy-MM-dd'T'HH:mm:ss.SSSZ\"}}}}}").unwrap();
```

Comment:
It wouldn't really, but if it's unique it can be used as an id field which would prevent Elasticsearch from generating one for the event anyway.